meta/language.LL1

```diff
@@ -0,0 +1,178 @@
+// Also the order of definitions and notation is currently not fully consistent
+
+// Stolen from https://webidl.spec.whatwg.org/#prod-i and adjusted to match Jakt more closely
+integer     = /-?([1-9][0-9]*|0[Xx][0-9A-Fa-f]+|0[Oo][0-7]*|0)([ui](8|16|32|64)|uz)?/
+float       = /-?(([0-9]+\.[0-9]*|[0-9]*\.[0-9]+)([Ee][+-]?[0-9]+)?|[0-9]+[Ee][+-]?[0-9]+)(f(64|32))?/
+            | /-?([1-9][0-9]*|0)f(64|32)/
+identifier  = /_?[A-Za-z][0-9A-Z_a-z-]*/
+
+String      = '"' EscapedContent '"'
+ByteString = "b'" EscapedContent "'"
+CChar       = "'" EscapedContent "'"
+
+EscapedContent = /([^\\]|\\.)*/
+
+boolean_op = /[<>]|[<>=!]=/
+           | "is"
+
+op  = /[+-*\/%|^&]/
+    | /[&|+-*\/]=/
+    | boolean_op
+    | /\?\?=?/ 
+lop = /++|--?|[*~]|/
+    | "not"
+    | "&raw"
+rop = /++|--/
+    | ("[" Expression "]")
+
+Constant         = integer
+                 | float
+                 | String
+                 | ByteString
+                 | CChar
+                 | "None"
+                 | "true" | "false"
+
+BuiltinType      = /f(32|64)/
+                 | /[iu](8|16|32|64)/
+                 | "usize"
+                 | /c_(int|char)/
+                 | "bool"
+                 | "String"
+                 | "void"
+
+TypeExpression   = BuiltinType ["?"]
+                 | {identifier [GenericSpecifier] "::" } identifier [GenericSpecifier] ["?"]
+                 | "[" TypeExpression "]" ["?"]
+                 | "[" TypeExpression ":" TypeExpression "]" ["?"]
+                 | "{ TypeExpression "}" ["?"]
+                 | "raw" TypeExpression  ["?"]
+                 | "weak" TypeExpression "?"
+
+GenericSpecifier = "<" TypeExpression {"," TypeExpression} ">"
+
+ArrayLiteral     = "[" [ArrayItem [ArrayItems]] "]"
+                 | "[" Expression ";" integer "]"
+ArrayItems       = "," ArrayItem [ArrayItems]
+ArrayItem        = Expression
+
+MapLiteral       = "[:]"
+                 | "[" MapItem [MapItems] "]"
+MapItems         = "," MapItem [MapItems]
+MapItem          = Expression ":" Expression
+
+SetLiteral       = "{" [SetItem [SetItems]] "}"
+SetItems         = "," SetItem [SetItems]
+SetItem          = Expression
+
+Range            = Expression ".." Expression
+
+MatchExpression  = "match" Expression "{" {MatchCase} "}"
+MatchCase        = MatchKey "=>" Statement
+MatchKey         = MatchKey "|" MatchKey
+                 | {identifier "::"} identifier ["(" {identifier [":" identifier]} ")"]
+                 | Constant
+                 | "else"
+
+IsExpression     = Expression "is" TypeExpression
+CastExpression   = Expression "as!" TypeExpression
+
+Expression       = Expression op Expression
+                 | lop Expression
+                 | Expression rop
+                 | "(" Expression ")"
+                 | Constant
+                 | ArrayLiteral
+                 | MapLiteral
+                 | SetLiteral
+                 | Range
```

Comment:
This technically can't be here as it causes left recursion (there's tonnes of this, I won't bother with the rest).